Stackoverflow while code editing

QuarksTools: 1.11.0.299
Intellij: 2022 CE
```
nternal error: java.lang.NullPointerException

java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.runTask(QuteLanguageClient.java:131)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.access$000(QuteLanguageClient.java:64)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient$1.run(QuteLanguageClient.java:106)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:442)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressInCurrentThread$10(CoreProgressManager.java:561)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:189)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:176)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressInCurrentThread(CoreProgressManager.java:561)
	at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:381)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$runAsBackground$2(QuteLanguageClient.java:103)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$runAsBackground$3(QuteLanguageClient.java:117)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1728)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException
	at com.redhat.devtools.intellij.qute.psi.internal.java.AbstractQuteTemplateLinkCollector.visitField(AbstractQuteTemplateLinkCollector.java:89)
	at com.intellij.psi.impl.source.PsiFieldImpl.accept(PsiFieldImpl.java:355)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.acceptChildren(SharedImplUtil.java:185)
	at com.intellij.psi.impl.source.JavaStubPsiElement.acceptChildren(JavaStubPsiElement.java:113)
	at com.intellij.psi.JavaRecursiveElementVisitor.visitElement(JavaRecursiveElementVisitor.java:44)
	at com.intellij.psi.JavaElementVisitor.visitClass(JavaElementVisitor.java:44)
	at com.redhat.devtools.intellij.qute.psi.internal.java.AbstractQuteTemplateLinkCollector.visitClass(AbstractQuteTemplateLinkCollector.java:110)
	at com.intellij.psi.impl.source.PsiClassImpl.accept(PsiClassImpl.java:464)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.acceptChildren(SharedImplUtil.java:185)
	at com.intellij.psi.impl.source.PsiFileImpl.acceptChildren(PsiFileImpl.java:754)
	at com.intellij.psi.JavaRecursiveElementVisitor.visitElement(JavaRecursiveElementVisitor.java:44)
	at com.intellij.psi.PsiElementVisitor.visitFile(PsiElementVisitor.java:35)
	at com.intellij.psi.JavaElementVisitor.visitJavaFile(JavaElementVisitor.java:340)
	at com.intellij.psi.impl.source.PsiJavaFileBaseImpl.accept(PsiJavaFileBaseImpl.java:480)
	at com.redhat.devtools.intellij.qute.psi.internal.java.QuarkusIntegrationForQute.codeLens(QuarkusIntegrationForQute.java:41)
	at com.redhat.devtools.intellij.qute.psi.QuteSupportForJava.codeLens(QuteSupportForJava.java:60)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getJavaCodelens$13(QuteLanguageClient.java:171)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:933)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getJavaCodelens$14(QuteLanguageClient.java:171)
	at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.runTask(QuteLanguageClient.java:127)
	... 23 more

```

--- a/src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteLanguageClient.java
+++ b/src/main/java/com/redhat/devtools/intellij/qute/lsp/QuteLanguageClient.java
@@ -129,6 +129,7 @@ public class QuteLanguageClient extends LanguageClientImpl implements QuteLangua
       } catch (IndexNotReadyException e) {
       } catch (Throwable t) {
         future.completeExceptionally(t);
+        done = true;
       }
     }
     if (!done) {

--- a/src/main/java/com/redhat/devtools/intellij/qute/psi/internal/java/AbstractQuteTemplateLinkCollector.java
+++ b/src/main/java/com/redhat/devtools/intellij/qute/psi/internal/java/AbstractQuteTemplateLinkCollector.java
@@ -86,8 +86,11 @@ public abstract class AbstractQuteTemplateLinkCollector extends JavaRecursiveEle
 	@Override
 	public void visitField(PsiField node) {
 		if (node.getType() instanceof PsiClassType) {
-			if (TEMPLATE_CLASS.equals(((PsiClassType) node.getType()).resolve().getQualifiedName())) {
-				processTemplateLink(node);
+			PsiClass clazz = ((PsiClassType) node.getType()).resolve();
+			if (clazz != null) {
+				if (TEMPLATE_CLASS.equals(clazz.getQualifiedName())) {
+					processTemplateLink(node);
+				}
 			}
 		}
 		super.visitField(node);

